File: coco/src/main/java/coco/ep/m7/Neg.java
package coco.ep.m7;

import coco.ep.Exp;

public interface Neg<FT> extends coco.ep.m6.Neg<FT>, coco.ep.m7.Exp<FT> {

    default Exp<FT> powby(Exp<FT> other) {
        return this.mult(convert(this.lit(1.0)).powby(getInner()), convert(getInner()).powby(convert(other)));
    }
}

File: coco_csharp/CoCo/EP/M7/Neg.cs
namespace EP.M7 {
    public interface Neg<FT> : Exp<FT>, EP.M6.Neg<FT> {
        EP.Exp<FT> Exp<FT>.powby(EP.Exp<FT> other) {
            return mult(convert(lit(1.0)).powby(getInner()), convert(getInner()).powby(convert(other)));
        }
    }
}